src/components/questions/QuestionList.jsx

```diff
@@ -1,30 +1,55 @@
 import React, {useState, useEffect, useRef} from 'react';
 import Question from './IndividualQuestion.jsx';
+import AddQuestion from './AddQuestion.jsx';
 import styles from './questions.module.css';
 
 const QuestionList = (props) => {
 
+
+  let allQuestions = props.data.results;
   let firstTwoQuestions = props.data.results.slice(0, 2);
 
-  let items = firstTwoQuestions.map((question, i) => {
+  let items = allQuestions.map((question, i) => {
     return (
       <Question question={question} key={i}/>
     )
   })
-
+  // This code sets the length of the displayed questions (default is 2)
+  const [displayedQuestions, setDisplayed] = useState(items.slice(0, 2));
+
+  // Each click extends list of questions by up to 2
+  let moreQuestionsClick = () => {
+    if (items.length > displayedQuestions.length) {
+      let length = displayedQuestions.length;
+      setDisplayed(items.slice(0, length + 2));
+    }
+  }
+
+  // This stores the height of the client screen
   const screenHeight = window.innerHeight;
+
+  // this is the hook for tracking the height of the questions list element
```

Comment:
I stumbled across another way to do this that might be simpler if you want to check it out at some point.